frontend/src/metabase/dashboard/components/Sidebar.jsx

```diff
@@ -9,6 +9,7 @@ const WIDTH = 384;
 function Sidebar({ onClose, onCancel, closeIsDisabled, children }) {
   return (
     <div
+      data-testid="sidebar"
```

Comment:
Are we not treating `data-testid` attrs as a code smell?

Reply:
Not necessarily.
Even the creator of React Testing Library (Kent C. Dodds) suggests using testids rather than css classes if nothing else is present.
Now, I agree with your point that we should improve a11y of our app (something I've been trying to push forward for a long time). It's not a light task, tbh. 

My concern is that without a proper planning and good preparation we can maybe create some bad a11y accommodations which are maybe worse than not having any.

If we start using more HTML5 elements, we could get the most out of both worlds (because many of them have implicit ariya roles associated with them).

[`aside`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/aside) comes to mind, but tbh I am not sure if that's the best element for this role. Would need to do a bit more research on the best practices.